diff --git a/meta-ros2-jazzy/conf/ros-distro/include/jazzy/ros-distro.inc b/meta-ros2-jazzy/conf/ros-distro/include/jazzy/ros-distro.inc
index 30eec58fe..1dfffbc1a 100644
--- a/meta-ros2-jazzy/conf/ros-distro/include/jazzy/ros-distro.inc
+++ b/meta-ros2-jazzy/conf/ros-distro/include/jazzy/ros-distro.inc
@@ -234,6 +234,9 @@ ROS_UNRESOLVED_DEP-python3-flake8-docstrings = "python3-flake8-docstrings"
 # Used by generated-recipes/ament-lint/ament-flake8_0.16.0-1.bb
 ROS_UNRESOLVED_DEP-python3-flake8-quotes = "python3-flake8-quotes"
 
+# Used by generated-recipes/automatika-embodied-agents/automatika-embodied-agents_0.3.2-1.bb
+ROS_UNRESOLVED_DEP-python3-httpx = "python3-httpx"
+ROS_UNRESOLVED_DEP-python3-platformdirs="python3-platformdirs"
 ROS_UNRESOLVED_DEP-meson-native = "meson-native"
 ROS_UNRESOLVED_DEP-python3-ply = "python3-ply"
 ROS_UNRESOLVED_DEP-python3-jinja2 = "python3-jinja2"
@@ -300,6 +303,10 @@ ROS_UNRESOLVED_DEP-libbluetooth-dev = "bluez5"
 
 ROS_UNRESOLVED_DEP-libdbus-dev = "dbus"
 
+ROS_UNRESOLVED_DEP-python3-transforms3d = "python3-transforms3d"
+
+ROS_UNRESOLVED_DEP-python3-tqdm = "python3-tqdm"
+
 ROS_UNRESOLVED_DEP-python3-distro = "python3-distro"
 
 ROS_UNRESOLVED_DEP-python3-img2pdf = "python3-img2pdf"
